statics/workflows/flow-latency.yaml

```diff
@@ -0,0 +1,62 @@
+---
+UUID: "85757b38-6e21-1dee-b42d-28d2442e1017"
+name: "FlowLatency"
+description: "Calculate flow latency between two interfaces"
+parameters:
+  - name: source
+    description: Source node
+    type: node
+  - name: destination
+    description: Destination node
+    type: node
+source: |
+    function FlowLatency(from, to) {
+      var capture = new Capture();
+        capture.GremlinQuery = "G.V().Has('TID', '" + from + "')";
+
+        var packetInjection = new PacketInjection();
+        packetInjection.Src = "G.V().Has('TID', '" + from + "')"
+        packetInjection.Dst = "G.V().Has('TID', '" + to + "')"
+        packetInjection.Type = "icmp4"
+        packetInjection.ICMPID = Math.floor(Math.random() * 1000);
+        packetInjection.Count = 1
+
+        return client.captures.create(capture).then(function (c) {
+            capture = c
+            return sleep(2000)
+        }).then(function () {
+            return client.G.V().Has("TID", from)
+        }).then(function (nodes) {
+            if (nodes[0].Metadata.Neutron && nodes[0].Metadata.Neutron.IPV4) {
+                packetInjection.SrcIP = nodes[0].Metadata.Neutron.IPV4[0]
+            }
+            if (nodes[0].Metadata.ExtID && nodes[0].Metadata.ExtID["attached-mac"]) {
+                packetInjection.SrcMAC = nodes[0].Metadata.ExtID["attached-mac"]
+            }
+            return client.G.V().Has("TID", to)
+        }).then(function (nodes) {
+            if (nodes[0].Metadata.Neutron && nodes[0].Metadata.Neutron.IPV4) {
+                packetInjection.DstIP = nodes[0].Metadata.Neutron.IPV4[0]
+            }
+            if (nodes[0].Metadata.ExtID && nodes[0].Metadata.ExtID["attached-mac"]) {
+                packetInjection.DstMAC = nodes[0].Metadata.ExtID["attached-mac"]
+            }
+            return client.packetInjections.create(packetInjection)
+        }).then(function () {
+            return sleep(2000)
+        }).then(function () {
+            return client.G.Flows().Has("ICMP.ID", packetInjection.ICMPID)
```

Comment:
you should filter on TIDs too : like G.V().Has('TID', from).Flows("ICMP.ID", <id>)
As other captures and packets could be saved in Skydive